### ISSUE TITLE
Accessibility issue - Datepicker should be closed when input is unfocused

### ISSUE THREAD

**Issue by Ke1sy** [Hacker0x01/react-datepicker#3561]
**Describe the bug**
When calendar is open and user unfocus the input by returning to previous focusable element with alt+tab, it should close. 

**To Reproduce**

1. Start the screen reader and open the date-picker with keyboard.
2. Press "alt + tab" to return to previous focusable element.

**Expected behavior**
Datepicker should be not visible.

**Screenshots**
![image](https://user-images.githubusercontent.com/26119263/163409143-657ed379-eb6a-4d5b-b58a-84c4ef301ecf.png)

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Chrome (v. 100.0.4896.75), Firefox

**Comment by sebastianvitterso** [comment]
> Press "alt + tab"

I made the same mistake (which is how I found your issue), but I think you mean "shift + tab". 

Issue still exists, and is very annoying when there are multiple date-pickers in a row (e.g. in a "from", "to" filter scenario)

**Comment by ConcernedHobbit** [comment]
Related to #3685, the component does not blur correctly when tabbing out

**Comment by sebastianvitterso** [comment]
I see that #4050 fixed this, so the issue should be marked as "Completed", not "Not planned".
